src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,29 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.dao.impl.CountryDaoImpl;
+import mate.academy.hibernate.relations.exception.DataProcessingException;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import mate.academy.hibernate.relations.util.HibernateUtil;
+import org.hibernate.SessionFactory;
 
 public class CountryServiceImpl implements CountryService {
+    private SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+    private CountryDao countryDao = new CountryDaoImpl(sessionFactory);
```

Comment:
You should not create a new instance of CountryDaoImpl here. It should be injected through the constructor. This is a violation of Dependency Inversion Principle.